infrastructure/comms/src/connection/net_address/mod.rs

```diff
@@ -41,51 +59,90 @@ pub enum NetAddress {
     I2P(I2PAddress),
 }
 
+impl NetAddress {
+    /// Returns true if the [`NetAddress`] is an IP address, otherwise false
+    pub fn is_ip(&self) -> bool {
+        match *self {
+            NetAddress::IP(_) => true,
+            _ => false,
+        }
+    }
+
+    /// Returns true if the [`NetAddress`] is a Tor Onion address, otherwise false
+    pub fn is_tor(&self) -> bool {
+        match *self {
+            NetAddress::Tor(_) => true,
+            _ => false,
+        }
+    }
+
+    /// Returns true if the [`NetAddress`] is an I2P address, otherwise false
+    pub fn is_i2p(&self) -> bool {
+        match *self {
+            NetAddress::I2P(_) => true,
+            _ => false,
+        }
+    }
+}
+
 impl FromStr for NetAddress {
     type Err = NetAddressError;
 
+    /// Parses a [`str`] into a [`NetAddress`]
     fn from_str(address: &str) -> Result<Self, Self::Err> {
```

Comment:
check that the order is correct, they do look fine from a quick glance

Reply:
The parsers are strict enough that it won't matter what order they are in (if the onion parser doesnt see .onion where it expected, the parse will fail, if the IP socket parser sees anything other than an IP address it will fail)